libraries/chain/controller.cpp

```diff
@@ -2056,43 +2056,58 @@ struct controller_impl {
       }
    } FC_CAPTURE_AND_RETHROW() } /// apply_block
 
+
+   block_state_ptr create_block_state( const block_id_type& id, const signed_block_ptr& b, const block_header_state& prev ) {
+      auto trx_mroot = calculate_trx_merkle( b->transactions );
```

Comment:
Will `calculate_trx_merkle` take long time on the `main` thread when called by `create_block_state`?

Reply:
`create_block_state` is never called from the main thread. I'll add a comment.